### PR TITLE
raftstore: implement get/set high priority background threads

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/12842

What's Changed:

```commit-message
implement get/set high priority background threads 
```

### Related changes

https://github.com/tikv/rust-rocksdb/pull/752

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tabokie — COMMENTED** [review]
What is this for?

**Comment by SpadeA-Tang** [comment]
> What is this for?

Used to adjust flush threads when cluster stop to boost flushing memtables.

File: components/engine_traits/src/tablet.rs [hunk 1]
```diff
@@ -136,13 +136,23 @@ impl TabletContext {
     }
 }
 
+pub trait SharedResources {
```
Comment (tabokie) [inline components/engine_traits/src/tablet.rs:139]: No need for this.

Comment (SpadeA-Tang) [reply components/engine_traits/src/tablet.rs:139]: How can get env from tablet registry?

Comment (tabokie) [reply components/engine_traits/src/tablet.rs:139]: `Env` can be accessed through `DBOptions`. You need to update rust-rocksdb's DBOptions and tikv's DbOptions trait.

File: src/config/configurable.rs [hunk 2]
```diff
@@ -176,4 +192,18 @@ impl ConfigurableDb for TabletRegistry<RocksEngine> {
             }
         })
     }
+
+    fn set_high_priority_background_threads(&self, n: i32, allow_reduce: bool) -> ConfigRes {
+        assert!(n > 0);
+        loop_registry(self, |cache| {
+            if let Some(latest) = cache.latest() {
+                if let Err(e) = latest.set_high_priority_background_threads(n, allow_reduce) {
```
Comment (tabokie) [inline src/config/configurable.rs:200]:
```suggestion
                latest.set_high_priority_background_threads(n, allow_reduce)?;
```

Comment (SpadeA-Tang) [reply src/config/configurable.rs:200]: It will try again and again.

**Review by tabokie — APPROVED** [review]

File: Cargo.lock [hunk 3]
```diff
@@ -385,7 +385,7 @@ dependencies = [
 [[package]]
 name = "azure_core"
 version = "0.12.0"
-source = "git+https://github.com/Azure/azure-sdk-for-rust#1fac7b8760f2610afd74269e3cc27e76eb33e6a9"
+source = "git+https://github.com/Azure/azure-sdk-for-rust#be63d1eb8bcd8f760376fb4b71ad8a140261947f"
```
Comment (overvenus) [inline Cargo.lock:388]: Why update `azure_core`?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: Cargo.lock (hunk 3 above)
Comment (SpadeA-Tang) [reply Cargo.lock]: fixed

**Review by overvenus — APPROVED** [review]

**Pull request merged**